CMakeLists.txt

```diff
@@ -39,3 +39,19 @@ include(modules/fetch-nlohmann.cmake)
 include(modules/external-mgcxx.cmake)
 add_executable(mgcxx_text_search_example mgcxx_text_search.cpp)
 target_link_libraries(mgcxx_text_search_example PRIVATE nlohmann_json::nlohmann_json mgcxx_text_search tantivy_text_search)
+
+# # https://www.jwlawson.co.uk/interest/2020/02/23/cmake-external-project.html
+# # https://coderefinery.github.io/cmake-workshop/fetch-content/
+# include(modules/external-aws-sdk-cpp.cmake)
+# set(AWSSDK_DIR "${CMAKE_BINARY_DIR}/libawscpp-proj")
+# list(APPEND CMAKE_PREFIX_PATH ${AWSSDK_DIR})
+# # TODO(gitbuda): Figure out a way how to combine ExternalProject and find_package
+# find_package(AWSSDK REQUIRED COMPONENTS s3)
+# message(STATUS "${AWSSDK_PLATFORM_DEPS} ${AWSSDK_LINK_LIBRARIES}")
+
+include(modules/fetch-aws-sdk-cpp.cmake)
+
+add_executable(aws_example aws.cpp)
+target_link_libraries(aws_example)
```

Comment:
todo
try to list all cmake targets
conan actually writes how to use the lib -> try out